src/components/bottom-bar/bottom-bar-registry-manager.ts

```diff
@@ -0,0 +1,42 @@
+export interface IconComponent {
+  registerComponent(): any;
+  getSvgIcon(): any;
+  getComponentText(): any;
+}
+
+export class BottomBarRegistryManager {
+  private _registry: Map<string, any>;
+
+  constructor() {
+    this._registry = new Map();
+  }
+
+  public register(componentName: string, componentIcon: any): void {
+    if (!this.getComponentItem(componentName)) {
+      this._registry.set(componentName, componentIcon);
+    }
+  }
+
+  public unregister(componentName: string): void {
+    if (this.getComponentItem(componentName)) {
+      this._registry.delete(componentName);
+    }
+  }
+
+  public getComponentItem(componentName: string): any {
+    return this._registry.get(componentName);
+  }
+
+  public clear(): void {
+    this._registry.clear();
+  }
+}
+
+export const bottomBarRegistryManager: string = 'bottomBarRegistryManager';
+
+export const registerToBottomBar = (componentName: string, player: any, getIconDtoCallback: () => any): void => {
```

Comment:
why the registration need to be exported as a separate function and not as a singleton service as we do in upperBarManager

Reply:
can't remember why Singleton wasn't a good option. i think in case we have multiple players on page..